apps/meteor/client/hooks/useRootUrlChange.ts

```diff
@@ -0,0 +1,69 @@
+import { useRole, useSetting, useSettingSetValue, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { useMutation } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { Roles } from '../../app/models/client';
+import UrlChangeModal from '../components/UrlChangeModal';
+import { imperativeModal } from '../lib/imperativeModal';
+import { isSyncReady } from '../lib/userData';
+
+export const useRootUrlChange = (userId: string) => {
+	const { t } = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+	const isAdmin = useRole('admin');
+
+	const currentUrl = location.origin + window.__meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
+	const siteUrl = useSetting('Site_Url', '');
+	const documentDomain = useSetting('Document_Domain', '');
+	const setSiteUrl = useSettingSetValue('Site_Url');
+
+	const siteUrlMutation = useMutation({
+		mutationKey: ['settings', 'Site_Url'],
+		mutationFn: async (url: string) => {
+			await setSiteUrl(url);
+			return { url };
+		},
+		onSuccess: ({ url }) => dispatchToastMessage({ type: 'success', message: `${t('Saved')}, new site url is: ${url}` }),
+	});
+
+	useEffect(() => {
+		if (!Roles.ready.get() || !isSyncReady.get()) {
+			return;
+		}
+
+		if (!isAdmin) {
+			return;
+		}
+
+		if (!siteUrl) {
+			return;
+		}
+
+		if (window.__meteor_runtime_config__.ROOT_URL.replace(/\/$/, '') !== currentUrl) {
+			const confirm = (): void => {
+				imperativeModal.close();
+				siteUrlMutation.mutateAsync(currentUrl);
```

Comment:
I think we can go with `mutate()` if you are not using the promise
```suggestion
				siteUrlMutation.mutate(currentUrl);
```